packages/native/src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,171 @@
+   *
+   * The Carousel will scroll to the next item after this delay is elapsed,
+   * unless the user is currently dragging the content.
+   *
+   * This delay will also reset whenever the user taps on an indicator bullet to
+   * manually change the carousel item displayed.
+   */
+  autoDelay?: number;
+  /**
+   * Additional props to pass to the outer container.
+   * This container is a Flex element.
+   */
+  containerProps?: FlexboxProps;
+  /**
+   * Additional props to pass to the ScrollView element.
+   * See: https://reactnative.dev/docs/scrollview
+   */
+  scrollViewProps?: React.ComponentProps<typeof HorizontalScrollView>;
+  /**
+   * Additional props to pass to the indicators container.
+   * This container is a Flex element.
+   */
+  slideIndicatorContainerProps?: FlexboxProps;
+}>;
+
+function Carousel({
+  activeIndex,
+  autoDelay,
+  containerProps,
+  slideIndicatorContainerProps,
+  scrollViewProps,
+  children,
+}: Props) {
+  const [init, setInit] = useState(false);
+  const [activeIndexState, setActiveIndexState] = useState(activeIndex);
+  const disableTimer = useRef(false);
+  const [resetTimer, setResetTimer] = useState({});
+  const dimensions = useRef<{
+    contentWidth: number;
+    contentHeight: number;
+  } | null>(null);
+  const slidesLength = React.Children.count(children);
+  const scrollRef = useRef<ScrollView | null>(null);
+
+  const fullWidth = 100 * slidesLength;
+  const itemWidth = !dimensions.current
+    ? 0
+    : dimensions.current.contentWidth / slidesLength;
+
+  const scrollToIndex = useCallback(
+    (index, animated = true) => {
+      if (scrollRef.current && dimensions.current) {
+        scrollRef.current.scrollTo({
+          x: itemWidth * index,
+          animated,
+        });
+      }
+    },
+    [itemWidth]
+  );
+
+  useEffect(() => {
+    // On init scroll to the active index prop location - if specified.
+    if (init && activeIndex) scrollToIndex(activeIndex, false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
+
+  useEffect(() => {
+    if (scrollToIndex && typeof activeIndex === "number") {
+      scrollToIndex(activeIndex);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeIndex]);
+
+  const onContentSizeChange = useCallback(
+    (contentWidth: number, contentHeight: number) => {
+      dimensions.current = { contentWidth, contentHeight };
+      setInit(true);
+    },
+    []
+  );
+
+  const onScroll = useCallback(
+    ({ nativeEvent: { contentOffset, contentSize } }) => {
+      const newIndex = Math.abs(
+        Math.round((contentOffset.x / contentSize.width) * slidesLength)
+      );
+      setActiveIndexState(newIndex);
+    },
+    [slidesLength]
+  );
+
+  useEffect(() => {
+    if (!autoDelay) return;
+
+    const interval = setInterval(() => {
+      if (!disableTimer.current) {
+        setActiveIndexState((index) => {
+          const newIndex = index ? (index + 1) % slidesLength : 0;
```

Comment:
Is it by design that this prevents automatically scroll when the current index is 0 ?

Reply:
This is actually a late bug introduced by trying to fix a typescript error, good catch!!